drivers/web/framework/httprequest/session.py

```diff
@@ -42,3 +44,17 @@ def session_maker(request: HttpRequest) -> Session:
         json_str = cookies[json_start_index:json_end_index + 1]
     session_data = json.loads(json_str)
     return Session(session_data)
+
+
+def auth_needed(session_need: str):
+    def decorator(f: Callable):
+        def wrapped(self, request: HttpRequest):
+            session = session_maker(request)
+            if session_need not in session:
+                session.invalidate()
+                return template_http_response("index.html",
```

Comment:
in the same way that we set the template directory in settings.py, on application, we can't set the default template to use here? Something like setting.AUTH_REDIRECT ?

Reply:
nice idea!